src/datastore/pg-store-v2.ts

```diff
@@ -629,12 +629,52 @@ export class PgStoreV2 extends BasePgStoreModule {
       if (cycleCheck.count === 0)
         throw new InvalidRequestError(`PoX cycle not found`, InvalidRequestErrorType.invalid_param);
       const results = await sql<(DbPoxCycleSigner & { total: number })[]>`
-        SELECT
-          signing_key, weight, stacked_amount, weight_percent, stacked_amount_percent,
-          COUNT(*) OVER()::int AS total
-        FROM pox_sets
-        WHERE canonical = TRUE AND cycle_number = ${args.cycle_number}
-        ORDER BY weight DESC, stacked_amount DESC, signing_key
+        WITH signer_keys AS (
+            SELECT DISTINCT ON (stacker) stacker, signer_key
+            FROM pox4_events
+            WHERE canonical = true AND microblock_canonical = true
+                AND name in ('stack-aggregation-commit-indexed', 'stack-aggregation-commit')
+                AND start_cycle_id <= ${args.cycle_number}
+                AND end_cycle_id >= ${args.cycle_number}
```

Comment:
Can this be an exact match for start and an exact match +1 for end, since the agg events will always target exactly one cycle? Might be better perf, either way the query might be more ~~readable~~understandable.

Reply:
That should be 
```sql
	AND start_cycle_id = ${args.cycle_number}
	AND end_cycle_id = (${args.cycle_number} + 1) -- this one probably isn't required?
```
right?

Reply:
This condition works for the aggregation events

Reply:
Perfect, thanks!